src/main/java/uk/gov/companieshouse/api/accounts/controller/CompanyAccountController.java

```diff
@@ -50,10 +53,26 @@ public ResponseEntity createCompanyAccount(@Valid @RequestBody CompanyAccount co
                 .getAttribute(AttributeName.TRANSACTION.getValue());
 
         String requestId = request.getHeader("X-Request-Id");
-        ResponseObject result = companyAccountService
-                .createCompanyAccount(companyAccount, transaction, requestId);
-        return apiResponseMapper
-                .map(result.getStatus(), result.getData(), result.getErrorData());
+
+        ResponseEntity responseEntity;
+        try {
+            CompanyAccount companyAccountInserted = companyAccountService
+                    .createCompanyAccount(companyAccount, transaction, requestId);
+            responseEntity = apiResponseMapper
+                    .map(ResponseStatus.SUCCESS_CREATED, companyAccountInserted);
+        } catch (ApiErrorResponseException aere) {
+            LOGGER.error(aere);
+            responseEntity = apiResponseMapper.map(ResponseStatus.TRANSACTION_PATCH_ERROR);
```

Comment:
My understanding from the meeting the other day was that we would no longer use ResponseStatus mappings for exceptional responses (i.e. 5XX errors) so this should explicitly be set as a 500 error in the controller without using the ResponseStatus.

The apiResponseMapper could be changed to map Exception types to http response codes instead so this would become:
`responseEntity = apiResponseMapper.map(aere);`
and the updated response mapper method would map an `ApiErrorResponseException` to a 500